features/discovery/src/main/java/org/opennms/netmgt/discovery/messages/DiscoveryJob.java

```diff
@@ -98,4 +103,18 @@ public String toString() {
                  .add("ranges", m_ranges)
                  .toString();
     }
+    
+    /**
+     * <P>
+     * Returns int : the total task timeout for all ip ranges
+     * </P>
+     */
+    public int calculateTaskTimeout(){
+    	BigInteger taskTimeOut = BigInteger.ZERO;
+        for(final IPPollRange range : m_ranges) {        	
+        	taskTimeOut = taskTimeOut.add((BigInteger.valueOf(range.getRetries()).add(BigInteger.ONE)).multiply(range.getAddressRange().getSizeOfIpAddrRange()).multiply(BigInteger.valueOf(range.getTimeout())).multiply(FUDGE_FACTOR.toBigInteger()));
```

Comment:
do you want me to perform all the calculations in BigDecimal or only for FUDGE_FACTOR field.